### DISCUSSION TITLE
Long running Tasks with KubernetesExecutor

### DISCUSSION THREAD

**Discussion by archit-goel01** [apache/airflow#36598] [Q&A]
I have a scenario, where I have weekly job which I want to execute , Job(Dag) has 2 tasks and I want first task to be running until second task is completed as first task is for running a docker image/application and second task wants to execute a method from this application.

Is it possible to achieve this using KubernetesExecutor or KubernetesPodOperator?

I know we have an option of CeleryKubernetesExecutor as well but in that case my worker pod will be sitting idle when no tasks are running.

what is the other option shall we keep both these tasks operation as one task itself?

**Comment by potiuk** [comment]
Setup/Teardown https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#setup-and-teardown

You have to have something where your application will be running and setup the application in well, setup, and tear it down after your main task is complete.

Available in Airflow 2.7.
